### ISSUE TITLE
Reporting undef errors in the LFE shell (versus Erlang shell).

### ISSUE THREAD

**Issue by mqsoh** [lfe/lfe#114]
I'm not sure my title is apt because I'm ignorant of both Erlang and LFE. This is what I'm observing. In `undef-no-error.lfe`:

```
(defmodule undef-no-error
  (export (start-link 0)))

(defun start-link []
  (supervisor:start_link (MODULE) '()))

(defun init [_] 'ignore)
```

I should have exported `init/1`! This is what I get in the LFE shell:

```
> (undef-no-error:start-link)
*** ERROR: Shell process terminated! (^G to start new job) ***
```

In the Erlang shell I get a more informative error:

```
1> 'undef-no-error':'start-link'().
** exception exit: undef
     in function  'undef-no-error':init/1
        called as 'undef-no-error':init([])
     in call from supervisor:init/1 (supervisor.erl, line 243)
     in call from gen_server:init_it/6 (gen_server.erl, line 306)
     in call from proc_lib:init_p_do_apply/3 (proc_lib.erl, line 237)
```

I have a muddled mind and spent some time thinking I'd passed incorrect arguments in my real module.

**Comment by rvirding** [comment]
The reason for the difference in behaviour between the LFE and the erlang shells is a bug in the LFE shell. :-( It does not properly handle getting exit signals from linked processes. I will fix it.

**Comment by arpunk** [comment]
@rvirding Is this from `lfe_shell:shell_eval/3`?

**Comment by rvirding** [comment]
This has actually been fixed in commits 988b6a7fbc04aff41c5645351ce729acbab4fe5d, 76745fe470e01f5a1ab9aa52952f9c849ac224d6 and db01a2b63b90e7ee47e46cb4b9121a38564833f1. My error in not connecting that to this issue. I will close it now.
